.buildkite/pipeline.yml

```
@@ -4,375 +4,437 @@ agents:
-    if: |
-      build.branch !~ /^gh-readonly-queue\/master/
-        && build.branch != "master"
-        && build.branch !~ /^release-candidate/
-        && build.branch != "rc-latest"
-    depends_on: linux-nix
-    key: trigger-build-windows-artifacts
-
-  - label: 'Build package (windows)'
-    key: windows-package
-    depends_on: [linux-nix, trigger-build-windows-artifacts]
-    command: nix build -o result/windows .#ci.artifacts.win64.release
-    artifact_paths: [ "./result/windows/**" ]
-    agents:
-      system: ${linux}
-    env:
-      TMPDIR: "/cache"
-
-  - label: 'Build testing bundle (windows)'
-    key: windows-testing-bundle
-    depends_on: [linux-nix, trigger-build-windows-artifacts]
-    command: nix build -o result/windows-tests .#ci.artifacts.win64.tests
-    artifact_paths: [ "./result/windows-tests/**" ]
-    agents:
-      system: ${linux}
-    env:
-      TMPDIR: "/cache"
-
-  - block: "Run E2E tests"
-    if: |
-      build.branch !~ /^gh-readonly-queue\/master/
-        && build.branch != "master"
-        && build.branch !~ /^release-candidate/
-        && build.branch != "rc-latest"
-    depends_on: linux-nix
-    key: trigger-e2e-tests
-
-  - label: "Run E2E tests"
-    key: e2e
-    depends_on: trigger-e2e-tests
-    command: 'nix develop --command bash -c "just e2e-local"'
-    agents:
-      system: ${linux}
-    env:
-      TMPDIR: "/cache"
-
-  - label: Manage rc-latest tags
+  - group: "Linux Setup"
+    key: "linux-setup"
+    steps:
+    - label: 'Check nix (linux)'
+      # Check whether regenerate.sh was applied when it had to be applied.
+      key: linux-nix
+      commands:
+        - './nix/regenerate.sh'
+      agents:
+        system: ${linux}
+
+    - block: "Cabal release"
+      if: |
+        build.branch !~ /^gh-readonly-queue\/master/
+            && build.branch != "master"
+            && build.env("RELEASE_CANDIDATE") == null
+      depends_on: linux-nix
+      key: cabal-release-block
+
+    - label: 'Cabal build all -frelease'
+      key: cabal-release
+      depends_on: cabal-release-block
+      command: |
+        nix develop -c cabal update
+        nix develop -c cabal build all -frelease
+      agents:
+        system: ${linux}
+
+    - label: 'Build benchmarks (linux)'
+      key: build-benchmarks
+      depends_on: linux-nix
+      command: 'nix build .#ci.benchmarks.all'
+      agents:
+        system: ${linux}
+      env:
+        TMPDIR: "/cache"
+
+    - label: 'Build package (linux)'
+      key: linux-package
+      depends_on:
+        - linux-nix
+      command: ./scripts/buildkite/main/linux-package.sh
+      artifact_paths: [ "./result/linux/**" ]
+      agents:
+        system: ${linux}
+
+    - label: 'Check Cabal Configure'
+      key: cabal-configure
+      depends_on: linux-nix
+      command: 'nix develop --command scripts/buildkite/check-haskell-nix-cabal.sh'
+      agents:
+        system: ${linux}
+
+  - group: "Linux Tests"
```

```suggestion
  - group: "Linux Checks"
```

Could you adjust to the nomenclature from "Continuous Integration :: AP"?